src/Readers/Vipr.Reader.OData.v4/OdcmReader.cs

```diff
@@ -518,47 +532,76 @@ private bool IsOperationBoundTo(IEdmOperation operation, IEdmEntityType entityTy
 
             private void WriteProperty(OdcmClass odcmClass, IEdmEntitySet entitySet)
             {
-                var odcmType = ResolveType(entitySet.EntityType().Name, entitySet.EntityType().Namespace);
-                var odcmProperty = new OdcmEntitySet(entitySet.Name)
+                try
                 {
-                    Class = odcmClass,
-                    IsCollection = true,
-                    IsLink = true
-                };
+                    var odcmType = ResolveType(entitySet.EntityType().Name, entitySet.EntityType().Namespace);
+                    var odcmProperty = new OdcmEntitySet(entitySet.Name)
+                    {
+                        Class = odcmClass,
+                        IsCollection = true,
+                        IsLink = true
+                    };
 
-                odcmProperty.Projection = new OdcmProjection
-                {
-                    Type = odcmType,
-                    BackLink = odcmProperty
-                };
-                
-                _propertyCapabilitiesCache.Add(odcmProperty, OdcmCapability.DefaultEntitySetCapabilities);
+                    odcmProperty.Projection = new OdcmProjection
+                    {
+                        Type = odcmType,
+                        BackLink = odcmProperty
+                    };
 
-                AddVocabularyAnnotations(odcmProperty, entitySet);
+                    try
+                    {
+                        _propertyCapabilitiesCache.Add(odcmProperty, OdcmCapability.DefaultEntitySetCapabilities);
+                    }
+                    catch (Exception e)
+                    {
+                        Logger.Error(e, e.Message);
+                    }
+
+                    AddVocabularyAnnotations(odcmProperty, entitySet);
 
-                odcmClass.Properties.Add(odcmProperty);
+                    odcmClass.Properties.Add(odcmProperty);
+                }
+                // If we hit an invalid type, skip this property
+                catch (Exception e)

```

Comment:
We might be overwhelmed with logging errors here. Could the logger have a level/scoping related to it so this is only logged under 'verbose'? If we feel this should only throw an exception when there is something missing that shouldn't be, we could log under 'error'.